Components/Config/PaymentConfigResolver.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+namespace MollieShopware\Components\Config;
+
+use MollieShopware\Components\Constants\OrderCreationType;
+use MollieShopware\Components\Constants\PaymentMethodType;
+use MollieShopware\Components\Translation\Translation;
+use MollieShopware\Exceptions\MolliePaymentConfigurationNotFound;
+use MollieShopware\Models\Payment\Configuration;
+use MollieShopware\Models\Payment\ConfigurationKeys;
+use MollieShopware\Models\Payment\Repository;
+use MollieShopware\MollieShopware;
+use Shopware\Components\Model\ModelManager;
+
+class PaymentConfigResolver
+{
+
+    /**
+     * @var ConfigFactory
+     */
+    protected $configFactory;
+
+    /**
+     * @var Repository
+     */
+    private $repoPaymentConfig;
+
+    /**
+     * @var Translation
+     */
+    private $translations;
+
+
+    /**
+     * PaymentConfigResolver constructor.
+     * @param ConfigFactory $configFactory
+     * @param ModelManager $models
+     * @param Translation $translations
+     */
+    public function __construct(ConfigFactory $configFactory, ModelManager $models, Translation $translations)
+    {
+        $this->configFactory = $configFactory;
+        $this->translations = $translations;
+
+        /** @var Repository $repoPayments */
+        $repoPayments = $models->getRepository(Configuration::class);
+        $this->repoPaymentConfig = $repoPayments;
+    }
+
+
+    /**
+     * Gets the final method type for the provided
+     * payment method in the provided shop.
+     * This is either the one from the payment configuration,
+     * or the global plugin setting if that has been configured.
+     *
+     * @param string $paymentMethod
+     * @param int $shopId
+     * @return int
+     * @throws MolliePaymentConfigurationNotFound
+     * @throws \Doctrine\DBAL\Exception
+     */
+    public function getFinalMethodType($paymentMethod, $shopId)
+    {
+        $fullPaymentMethod = $this->getPaymentFullName($paymentMethod);
+        $cleanPaymentMethod = str_replace(MollieShopware::PAYMENT_PREFIX, '', $paymentMethod);
+
+
+        # fetch the configuration for this payment method
+        $pluginConfig = $this->configFactory->getForShop($shopId);
+        $paymentConfig = $this->repoPaymentConfig->getByPaymentName($fullPaymentMethod);
+
+        # get any translated values
+        # for our provided shop
+        $translatedValue = $this->translations->getPaymentConfigTranslation(
+            ConfigurationKeys::METHODS_API,
+            $paymentConfig->getPaymentMeanId(),
+            $shopId
+        );
+
+        # use either the snippet or the
+        # value from the config directly
+        $methodType = (!empty($translatedValue)) ? (int)$translatedValue : (int)$paymentConfig->getMethodType();
```

Comment:
Why is it a int cast? Maybe better name methodTypeId or something like that? I would await string